src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,154 @@
+package com.google.sps.servlets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.cloud.language.v1.AnalyzeSentimentResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class DataServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private DataServlet newTermRating;
+  private Entity termEntity;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    languageService = Mockito.mock(LanguageServiceClient.class);
+    newTermRating = new DataServlet(languageService);
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+  @Mock LanguageServiceClient languageService;
+
+  @Test
+  public void addTermRating_newRatingNoTranslation() throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    request = Mockito.mock(HttpServletRequest.class);
+    when(request.getParameter("term-input")).thenReturn("I do not like this.");
+    when(request.getParameter("rating-term")).thenReturn("1");
+    when(request.getParameter("hoursOfWork")).thenReturn("8");
+    when(request.getParameter("difficulty")).thenReturn("4");
+    when(request.getParameter("prof-input")).thenReturn("The professor was amazing.");
+    when(request.getParameter("rating-professor")).thenReturn("3");
+    when(request.getParameter("ID")).thenReturn("numberOneId");
+
+    Entity termEntity = new Entity("Term");
```

Comment:
Suggestion: you can move this block of code up above the block of "whens" above so that the "when" just below this can join the others.